src/librbd/deep_copy/ImageCopyRequest.cc

```diff
@@ -258,13 +261,7 @@ void ImageCopyRequest<I>::handle_object_copy(uint64_t object_no, int r) {
       }
     }
 
-    while (true) {
-      r = send_next_object_copy();
-      if (r != 1) {
-        break;
-      }
-    }
-
+    send_next_object_copy();
```

Comment:
Ah, one more minor thing. The commit e5a21e9 changed `send_next_object_copy` to return `int` instead of void. Now when the return value is not used I think we can revert that part too?

Reply:
It is used as a bool to break out of the "send `rbd_concurrent_management_ops` initial requests" loop in `send_object_copies`.

Reply:
Indeed. Though it seems like it would not hurt much if we unconditionally made `rbd_concurrent_management_ops` initial calls of `send_next_object_copy`, even if some of them were unnecessary and just returned. But I am ok with the current version.

Reply:
Ah, never mind. I see there is a complication and the return value is useful there.

Reply:
No, I think you were right about "complication" ;)  Consider an image with less than `rbd_concurrent_management_ops` objects.  If `send_next_object_copy` was `void`, the loop would never end because `m_current_ops` would never reach `max_ops`.